File: 3day/ALU-main/ALU-main/java/ALUTest.java
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class ALUTest {

    @Test
    void enableSignal() {
        ALU alu = new ALU();
        alu.setOperand1(10);
        alu.setOperand2(20);
        alu.setOPCODE("ADD");

        Result ret = new Result();
        alu.enableSignal(ret);

        assertEquals(30, ret.getResult());
        assertEquals(0, ret.getStatus());
    }
}

File: 3day/ALU-main/ALU-main/cpp/ALU_test.cpp
#include "pch.h"
#include "../Project6/ALU.cpp";
#include "../Project6/Result.cpp";

TEST(ALU, ADDTest) {
	ALU alu;
	alu.setOperand1(10);
	alu.setOperand2(20);
	alu.setOPCODE("ADD");

	Result ret;
	alu.enableSignal(&ret);

	EXPECT_EQ(30, ret.getResult());
	EXPECT_EQ(0, ret.getStatus());
}
